### PR TITLE
Commenting EventIO methods

### PR DESCRIPTION
**Pull request by a-canary**
with doxygen style comments

### CODE REVIEW

File: iocore/net/P_UnixNet.h [hunk 1]
```diff
@@ -98,14 +100,37 @@ struct EventIO {
...
+   */
   int refresh(int events);
+
+  /** Remove the kernal or epoll event.
+     @return int: 0 on success, -1 on error
+   */
   int stop();
+
+  /** Remove the epoll event and close the connection.
+     @return int: 0 on success, -1 on error
```
Comment (SolidWallOfCode) [inline iocore/net/P_UnixNet.h:130]: Don't put the return type in the `@return` - Doxygen knows the return type and will handle that for you.

File: iocore/net/P_UnixNet.h [hunk 2]
```diff
@@ -98,14 +100,37 @@ struct EventIO {
...
+
+  /** Alter the events that will trigger the continuation, for level triggered I/O.
+     @param events: add with positive mask(+EVENTIO_READ), or remove with negative mask (-EVENTIO_READ)
+     @return int: the number of events created, -1 is error
+   */
   int modify(int events);
-  // Refresh the existing events (i.e. KQUEUE EV_CLEAR), for edge triggered I/O
+
+  /** Refresh the existing events (i.e. KQUEUE EV_CLEAR), for edge triggered I/O
+     @param events: mask of events
```
Comment (SolidWallOfCode) [inline iocore/net/P_UnixNet.h:119]: Don't use colon as a separator. Space suffices.

File: iocore/net/P_UnixNet.h [hunk 3]
```diff
@@ -506,9 +531,7 @@ check_transient_accept_error(int res)
   }
 }
 
-//
-// Disable a NetEvent
-//
+/// Disable the read epoll on a NetEvent
```
Comment (SolidWallOfCode) [inline iocore/net/P_UnixNet.h:534]: This needs a lot more detail.
```
/** Disable reading on the NetEvent @a ne
     @param nh Nethandler that owns @a ne
     @param ne The @c NetEvent to modify.

     - If write is already disable, also disable the inactivity timeout.
     - Remove the @c epoll READ flag.
     - Take @a ne out of the read ready list.
*/ 
```

**Comment by zwoop** [comment]
Need this into 9.0.0, to avoid merge conflicts from other PRs that needs to go there.

Cherry-picked to v9.0.x branch.

**Pull request merged**